### PR TITLE
Add new definition for jiff

### PR DESCRIPTION
**Pull request by daniel-kneip**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## jiff ([<kbd>unpkg</kbd>](https://unpkg.com/browse/jiff@latest/))
was missing the following properties:
1. diff
1. patch
1. patchInPlace
1. inverse
1. clone

<details>
<summary>as well as these 3 other properties...</summary>
<p>InvalidPatchOperationError, TestFailedError, PatchNotInvertibleError</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4d05c46f079bdf5628f8308e6e152e46f0b757ce
</p>

File: types/jiff/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+     *
+     * @returns patched version of x. If x is
+     *  an array or object, it will be mutated and returned. Otherwise, if
+     *  x is a value, the new value will be returned.
+     */
+    patch: (changes: JSONPatch, x: JSONValue, options?: PatchOptions) => JSONValue;
+    patchInPlace: (patch: JSONPatch, a: JSONValue) => void;
+};
+
+export default jiff;
```
Comment (gabritto) [inline types/jiff/index.d.ts:64]: I think you should use `export = jiff` here instead, to match how the exports are handled in the original package.

**Review by gabritto — CHANGES_REQUESTED** [review]

File: types/jiff/index.d.ts (hunk 1 above)
Comment (daedal-knickerbockers) [reply types/jiff/index.d.ts]: Thanks for the review! I made the necessary changes, but somehow messed up my branch by trying to merge in the master branch and now test-all script... I'll fix this and create a new pull-request

**Comment by daedal-knickerbockers** [comment]
I messed up the branch by pulling in the master, will resolve this and create a new pull request - sorry!

**Pull request closed without merging**